### DISCUSSION TITLE
Internal BlueRetro and use of original controllers

### DISCUSSION THREAD

**Discussion by pkerga** [darthcloud/BlueRetro#196] [Q&A] [answered]
Hello
I don't think the subject has already been discussed: once the BlueRetro is installed internally, is it still possible to use the original controllers?
- For the N64, I can say that the controllers work without any problem.
- it seemed to me that once installed in the Dreamcast, there was no longer any way to use the controllers (which makes the planned integration of the VMU memory all the more interesting)
- I saw a video and it seems that the use of joysticks on GC remains possible.

Do you have more information ?

**Comment by darthcloud** [comment, marked as answer]
No you can't, you need to either put a switch on ESP power or on each data line.

If it work it due to pure luck and likely unreliable.

Internal mod are not officially supported.

I might at some point add official support but not short term.

**Reply by MikeHandorf** [reply]
If I were to install a relay (like we used to do for internal XBOX ir dongle) on the data line **in theory**, this would disable the blueretro on the port that has the controller plugged in. The question is then, would the esp32 know that it's disconnected and not assign a controller to that port? for example, I plug in a controller on port 2 and the relay disables the data line on port 2, would the blueretro know to assign the first bluetooth device to port 1, the second to port 3, the third to port 4?

**Reply by darthcloud** [reply]
Not currently, but I do plan to add support for this in the future.

**Reply by MikeHandorf** [reply]
Very cool. I currently have some relays on order. I'll give it a try and update my post when they arrive. In the meantime I'll just add a switch.
